### PR TITLE
8036 duplicate message tracking tests for entity emailer

### PR DESCRIPTION
**Pull request by argush3**
*Issue #:* /bcgov/entity#8036

*Description of changes:*

* Added tests to verify message tracking works for following test scenarios:
  * successfully track processed message
  * does not try to re-process processed message
  * successfully track failure when exception thrown
  * failed message will succeed on retry if original error was a send email exception and email exception no longer thrown
  * retry count is properly tracked

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

File: queue_services/entity-emailer/tests/conftest.py [hunk 1]
```diff
@@ -102,11 +117,44 @@ def client_id():
 
 
 @pytest.fixture(scope='session')
-def db(app):  # pylint: disable=redefined-outer-name, invalid-name
+def db(app, tracker_app):  # pylint: disable=redefined-outer-name, invalid-name
```
Comment (thorwolpert) [inline queue_services/entity-emailer/tests/conftest.py:120]: tracker should be its own fixture

**Review by thorwolpert — CHANGES_REQUESTED** [review]
tracker should be in its own fixture

File: queue_services/entity-emailer/tests/conftest.py (hunk 1 above)
Comment (argush3) [reply queue_services/entity-emailer/tests/conftest.py]: moved tracker_db to own fixture

**Comment by argush3** [comment]
@thorwolpert ready for another look.  thanks!

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**